line 88 in utils will crash under python3, which the dict no longer have iteritems().
